atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/internalschema/ReadOnlyTransactionSchemaManager.java

```diff
@@ -37,8 +33,8 @@ public ReadOnlyTransactionSchemaManager(CoordinationService<InternalSchemaMetada
     public Integer getTransactionsSchemaVersion(long timestamp) {
         if (timestamp < AtlasDbConstants.STARTING_TS) {
             throw new SafeIllegalStateException(
-                    "Query attempted for timestamp {} which was never given out by the"
-                            + " timestamp service, as timestamps start at {}",
+                    "Query attempted for timestamp (queriedTimestamp) which was never given out by the"
+                            + " timestamp service, as timestamps start at (startOfTime).",
```

Comment:
I think this reads a bit weirdly; maybe just go for:
```suggestion
                    "Query attempted for a timestamp which was never given out by the"
                            + " timestamp service as the timestamp is earlier than the starting timestamp",
```

In other words, divorcing the message from the safe args while still making it clear what they mean.

Reply:
yeah, amplified because `startOfTime` is a slightly weird name (maybe I've been watching too many Brian Cox programmes)